Add job to rename the ExplorationOpportunitySummaryModel properties

DubeySandeep opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

DubeySandeep committed:
Add job to renmae the opportunity properties.

DubeySandeep commented on core/domain/opportunity_domain_test.py:
```diff
@@ -305,45 +306,45 @@ def test_invalid_lang_code_in_incomplete_translation_langs_fails_validation(
     def test_invalid_lang_code_in_assigned_voice_artist_langs_fails_validation(
             self):
-        self.valid_exp_opp_summary.assigned_voice_artist_in_language_codes = [
+        self.valid_exp_opp_summary.language_codes_with_assigned_voice_artists = [ # pylint: disable=line-too-long
```
Find a way to remove pragma here (ditto below)

DubeySandeep replied:
Done!

DubeySandeep commented on core/domain/opportunity_jobs_one_off.py:
```diff
@@ -84,3 +86,57 @@ def map(skill_model):
+            model_values = opportunity_model._values    # pylint: disable=protected-access
+
+            # This check blocks newly created models or rerunning this job on
+            # renamed models.
```
Incorrect comment after `or...`

DubeySandeep replied:
Done!

DubeySandeep commented on core/domain/opportunity_jobs_one_off.py:
```diff
@@ -84,3 +86,57 @@ def map(skill_model):
+            opportunity_model.language_codes_needing_voice_artists = old_value
+            yield ('SUCCESS_RENAMED[need_voice_artist_in_language_codes]', 1)
+
+        opportunity_model.update_timestamps(update_last_updated_time=False)
+        opportunity_model.put()
```
Don't update the model if nothing is changed!

DubeySandeep replied:
Done!

DubeySandeep committed:
Fix self-review comments.

seanlip commented:
Thanks @DubeySandeep! Looks fine to me, feel free to file a job run request for testing.

DubeySandeep commented:
Hi @seanlip, I've submitted the form to run/test the job on the backup server!

seanlip commented:
@DubeySandeep -- in this PR, could you please remove `third_party/python_libs/google/api_core/` from skip_files in app_dev.yaml? Keeping it in breaks production and led to a 500 error when I was first trying to deploy this branch. Thanks!

seanlip commented:
Also, I left a comment on your doc; once you've addressed it, I'll continue the testing process.

DubeySandeep committed:
Merge branch 'develop' of github.com:oppia/oppia into fix-field-names

DubeySandeep committed:
Removes some skip files from app_dev.yml

DubeySandeep commented:
> @DubeySandeep -- in this PR, could you please remove `third_party/python_libs/google/api_core/` from skip_files in app_dev.yaml? Keeping it in breaks production and led to a 500 error when I was first trying to deploy this branch. Thanks!

Done! (i.e, I've removed the path from skip_files in app_dev.yml)

> Also, I left a comment on your doc; once you've addressed it, I'll continue the testing process.

I've addressed the comment in the doc! (Done!)

@seanlip, PTAL!

DubeySandeep commented on app_dev.yaml:
```diff
@@ -306,7 +306,6 @@ skip_files:
 - third_party/python_libs/oauthlib/
 - third_party/python_libs/requests_oauthlib/
 - third_party/python_libs/uritemplate/
-- third_party/python_libs/google/api_core/
```
Removed this path as per request in [this comment](https://github.com/oppia/oppia/pull/11093#issuecomment-720003134).

> Sean Lip: @DubeySandeep -- in this PR, could you please remove third_party/python_libs/google/api_core/ from skip_files in app_dev.yaml? Keeping it in breaks production and led to a 500 error when I was first trying to deploy this branch. Thanks!

seanlip reviewed (approved):
Approved for run in production. Thanks!

vojtechjelinek reviewed (approved):
LGTM!

sagangwee reviewed (approved):
@DubeySandeep LGTM. Thanks!
